### ISSUE TITLE
Training loss

### ISSUE THREAD

**Issue by itera-del** [SadilKhan/Text2CAD#44]
I trained for 50 epochs and used 10,000 samples, but I found that Seq_loss converged too quickly and Seq Accuracy(Val) was higher than Seq Accuracy(Train). Is this correct?

![Image](https://github.com/user-attachments/assets/3834ef4e-1c00-4e8d-92ba-18a52479ad89)

**Comment by itera-del** [comment]
I want to make some optimizations to some of the model's structures, but I don't have a very good GPU, so I will take 10,000 samples and train for 50 epochs as a benchmark to compare and determine whether my improvements are effective. Do you think this is appropriate? In practice, I found that the curve has some problems

**Comment by W-Vicky11** [comment]
I also have a similar problem ,the Val Accuracy becomes lower and lower. And I don't konw if it is because my dataset is too small. In order to avoid the OOM problem, I only used 70,000 samples, and  it seems that it finally leads to the overfitting.

<img width="561" alt="Image" src="https://github.com/user-attachments/assets/ba124a05-b7e8-46a2-95dd-44c1f4ad1363" />

<img width="563" alt="Image" src="https://github.com/user-attachments/assets/bc713eef-49da-40e0-b2d7-4e583659dfac" />

**Comment by itera-del** [comment]
How many samples are there in the training set you loaded? Why do I only have about 110,000?

**Comment by W-Vicky11** [comment]
> How many samples are there in the training set you loaded? Why do I only have about 110,000?

There are nearly 160,000 samples in the train subset. 

![Image](https://github.com/user-attachments/assets/1972806b-effb-4e3a-9fcb-3c45d179d549)

I think you can check your train_test_val.json file or check your code in the location I show you in the below screenshot.

<img width="340" alt="Image" src="https://github.com/user-attachments/assets/b1657511-7eda-44a4-bb51-9b87c5c0fa24" />

**Comment by itera-del** [comment]
Our questions have not been answered yet

**Comment by SadilKhan** [comment]
HI,
Training on 10k samples will definitely overfit the model.
